src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/SquareRepositoryIT.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.cosmosdb.repository;
+
+import com.microsoft.azure.spring.data.cosmosdb.common.TestUtils;
+import com.microsoft.azure.spring.data.cosmosdb.domain.inheritance.Square;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestRepositoryConfig.class)
+public class SquareRepositoryIT {
+    @Autowired
+    SquareRepository repository;
+
+    private Square square1 = new Square("id_1", 1, 1);
+    private Square square2 = new Square("id_2", 2, 4);
+
+    @Before
+    public void setup() {
+        repository.save(square1);
+        repository.save(square2);
+    }
+
+    @After
+    public void cleanup() {
+        repository.deleteAll();
+    }
+
+    @Test
+    public void  testFindAll() {
+        final List<Square> result = TestUtils.toList(repository.findAll());
+
+        assertThat(result.size()).isEqualTo(2);
+    }
+
+    @Test
+    public void testFindIncludeInheritedFields() {
+        final Optional<Square> result = repository.findById(square1.getId());
+
+        assertThat(result.get()).isNotNull();
+        assertThat(result.get().getId().equals(square1.getId()));
+        assertThat(result.get().getLength()).isEqualTo(square1.getLength());
+        assertThat(result.get().getArea()).isEqualTo(square1.getArea());
```

Comment:
Better to use junit `Assert.assertEquals`, that is for UT.

Reply:
Could not understand correctly, this is IT.

Reply:
Sorry for misleading, here I mean `org.junit.Assert` is designed for test, better to chose this instead of `org.assertj.core.api.Assertions.assertThat`.

Reply:
OK, `AssertJ` is also for testing.

https://github.com/joel-costigliola/assertj-core

`AssertJ provides a rich and intuitive set of strongly-typed assertions to use for unit testing (with JUnit, TestNG or any other test framework).`

Reply:
OK, good to know that.